backend/app/main.py

```diff
@@ -1 +1,22 @@
-#
+from collections.abc import AsyncIterator
+from contextlib import asynccontextmanager
+
+from fastapi import FastAPI
+
+from app.anti_recommendation_engine import AntiRecommendationEngine
+from app.routers.routers import router
+
+
+@asynccontextmanager
+async def lifespan(app: FastAPI) -> AsyncIterator[None]:
+    """
+    A lifespan event to persist the AntiRecommendationEngine on start up.
+    """
+
+    app.state.anti_recommendation_engine = AntiRecommendationEngine()
+
+    yield
```

Comment:
comment: I didn't realize `yield` has the same semantics as `return`, where the `None` is implied. Interesting.